src/PuliServiceProvider.php

```diff
@@ -55,11 +55,17 @@ public function register(Application $app)
     public function boot(Application $app)
     {
         if (isset($app['twig']) && $app['puli.enable_twig']) {
-            $app['twig.loader'] = $app->share(function (Application $app) {
-                return new PuliTemplateLoader($app['puli.repository']);
+            $app->extend('twig.loader', function ($twigLoader, $app) {
+                $twigLoader->addLoader(new PuliTemplateLoader($app['puli.repository']));
```

Comment:
This depends on the fact that we get a chain loader here, no? What if that's not the case? Should we handle that?